library/common/http/client.cc

```diff
@@ -331,29 +305,45 @@ void Client::DirectStream::saveLatestStreamIntel() {
   }
   stream_intel_.stream_id = static_cast<uint64_t>(stream_handle_);
   stream_intel_.attempt_count = info.attemptCount().value_or(0);
-  saveFinalStreamIntel();
 }
 
 void Client::DirectStream::saveFinalStreamIntel() {
-  const auto& info = request_decoder_->streamInfo();
-  latency_info_.request_start_ms = std::chrono::duration_cast<std::chrono::milliseconds>(
-                                       info.startTimeMonotonic().time_since_epoch())
-                                       .count();
-  latency_info_.sent_byte_count = info.bytesSent();
-  latency_info_.received_byte_count = info.bytesReceived();
-  StreamInfo::TimingUtility timing(info);
-  setFromOptional(latency_info_.request_end_ms, timing.lastDownstreamRxByteReceived(),
-                  latency_info_.request_start_ms);
-  setFromOptional(latency_info_.dns_start_ms,
+  if (!request_decoder_) {
```

Comment:
I don't think the request decoder gets nulled out.  I think the canonical way to see if a stream is closed is to check
parent_.getStream(stream_handle_, ALLOW_ONLY_FOR_OPEN_STREAMS)